src/main.c

```diff
@@ -109,6 +109,7 @@ int     nlch = '\n';
 
 bool    tablesext, tablesverify, gentables;
 char   *tablesfilename=0,*tablesname=0;
+bool    is_tablesfilename_alloc;
```

Comment:
I said before, in the commit message, "Additionally, --tables-file=FILE will set tablesfilename to the argument **which should not be freed**, so this case is guarded against".

So there are three possible states for `tablesfilename`:
1. `NULL`
2. assigned from `--tables-file` command-line argument, which should be left alone because freeing it is undefined
3. strdup() of the `%option tables-file` value, which should be freed.
 
My fix is one reasonable solution.  Another would be to strdup() the command-line argument.